internal/provider/resource_custom_location.go

```diff
@@ -0,0 +1,211 @@
+package cpln
+
+import (
+	"context"
+	"fmt"
+
+	client "github.com/controlplane-com/terraform-provider-cpln/internal/provider/client"
+
+	"github.com/hashicorp/terraform-plugin-sdk/v2/diag"
+	"github.com/hashicorp/terraform-plugin-sdk/v2/helper/schema"
+)
+
+var allowedProviders = []string{"byok"}
+
+func resourceCustomLocation() *schema.Resource {
+	return &schema.Resource{
+		CreateContext: resourceCustomLocationCreate,
+		ReadContext:   resourceCustomLocationRead,
+		UpdateContext: resourceCustomLocationUpdate,
+		DeleteContext: resourceCustomLocationDelete,
+		Schema: map[string]*schema.Schema{
+			"cpln_id": {
+				Type:        schema.TypeString,
+				Description: "The ID, in GUID format, of the custom location.",
+				Computed:    true,
+			},
+			"name": {
+				Type:         schema.TypeString,
+				Description:  "Name of the custom location.",
+				ForceNew:     true,
+				Required:     true,
+				ValidateFunc: NameValidator,
+			},
+			"description": {
+				Type:             schema.TypeString,
+				Description:      "Description of the custom location.",
+				Optional:         true,
+				ValidateFunc:     DescriptionValidator,
+				DiffSuppressFunc: DiffSuppressDescription,
+			},
+			"tags": {
+				Type:        schema.TypeMap,
+				Description: "Key-value map of resource tags.",
+				Elem: &schema.Schema{
+					Type: schema.TypeString,
+				},
+				Optional:     true,
+				ValidateFunc: TagValidator,
+			},
+			"cloud_provider": {
+				Type:        schema.TypeString,
+				Description: "Cloud Provider of the custom location.",
+				Required:    true,
+				ForceNew:    true,
+				ValidateFunc: func(val interface{}, key string) (warns []string, errs []error) {
+					v := val.(string)
+					valid := false
```

Comment:
Remove this flag `valid` and let's do `return` on line 61. You also need to remove the if statement at line 64.